pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -159,7 +159,7 @@ func getAzureRoutes(cloud string, jsonData json.RawMessage) (map[string]types.Az
 }
 
 type azDatasourceExecutor interface {
-	ExecuteTimeSeriesQuery(ctx context.Context, originalQueries []backend.DataQuery, dsInfo types.DatasourceInfo, client *http.Client, url string, tracer tracing.Tracer) (*backend.QueryDataResponse, error)
+	ExecuteTimeSeriesQuery(ctx context.Context, originalQueries []backend.DataQuery, dsInfo types.DatasourceInfo, client *http.Client, url string, tracer tracing.Tracer, originalHTTPHeaders map[string]string) (*backend.QueryDataResponse, error)
```

Comment:
to simplify the interface, I would rather just pass down the `req *backend.QueryDataRequest`:

```suggestion
	ExecuteTimeSeriesQuery(ctx context.Context, req *backend.QueryDataRequest, dsInfo types.DatasourceInfo, client *http.Client, url string, tracer tracing.Tracer) (*backend.QueryDataResponse, error)
```